src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx

```diff
@@ -163,11 +163,25 @@ export const CheckoutPayment = ({
   const [getStatus] = useCheckoutStatusLazyQuery({
     pollInterval: 1000,
   })
-  const { search: is3DsComplete } = useLocation<{ search: string }>()
+  const history = useHistory()
+  const {
+    location: { search },
+  } = history
+  const is3DsError = search.includes('error')
+  const is3DsComplete = search.includes('3dsSuccess')
+
   const [isDataLoading, setIsDataLoading] = useState(false)
   const [isPageLoading, setIsPageLoading] = useState(false)
   const [isError, setIsError] = useState(false)
 
+  //handle 3ds error redirect
+  useEffect(() => {
+    if (is3DsError) {
+      history.replace('?')
+      setIsError(true)
```

Comment:
I'm not sure if it's a good idea to reuse this state.

Could we just use `is3DsError` instead?